repos/spack_repo/builtin/packages/emacs/package.py

```diff
@@ -127,6 +129,9 @@ def configure_args(self):
         args.extend(self.with_or_without("tree-sitter", variant="treesitter"))
         args.extend(self.with_or_without("json"))
 
+        if self.spec.satisfies("~sqlite"):
+            args.append("--with-sqlite3=no")
+
```

Comment:
```suggestion
        args.extend(self.with_or_without("sqlite3")
```

We should be able to continue using `with_or_without()` here.

Reply:
I do not think so:
<img width="697" height="294" alt="image" src="https://github.com/user-attachments/assets/4e2f5530-6c18-4382-aa2d-5dac18360e65" />
reddit user in [this discussion](https://www.reddit.com/r/emacs/comments/123klvf/emacs29_build_from_git_with_sqlite3/) says that `--with-sqlite3=no` exists, but `with-sqlite3=yes` does not exist.

The build options are:
- nothing -> build with sqlite support if sqlite3 available
- `--with-sqlite3=no` -> build without sqlite support

Reply:
In this case I think we could just drop the `--with-sqlite3=no` and have the presence / absence of the dependency controlled by the variant do the logic for us.

Reply:
done!
I removed the commit that added `--with-sqlite3=no`, squashed the correction `@29:` -> `@29.1:`, and rebased on develop